src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java

```diff
@@ -757,7 +757,21 @@ public void init(AdaptorContext context) throws Exception {
               sidLookupHost,sidLookupPort, sidLookupUsername,
               sidLookupPassword,sidLookupMethod);     
       }
-      
+      // Web Application which does not have root site is not supported 
+      // by SharePoint. Adaptor returns Error: Document not found (404) 
+      Holder<String> rootSite = new Holder<String>();
+      Holder<String> rootWeb = new Holder<String>();
+      long result = sharePointSiteDataClient.getSiteAndWeb(
+          configuredSharePointUrl.getVirtualServerUrl(), rootSite, rootWeb);
+      if (result != 0) {
+        log.log(Level.WARNING, "Ensure web application has root site created."
```

Comment:
There are missing spaces in the message at the line breaks, and no placeholder for the getVirtualServerUrl() argument, but I don't think the extra text here is useful, and so this is a log-and-throw anti-pattern. I would just include the URL in the exception text and drop the logging statement.